webapps/trunx-docs/app/form/general/CompleteFormExample.tsx

```diff
@@ -0,0 +1,180 @@
+  return (
+    <PageSection title="Colors">
+      <Columns>
+        <Column size="half">
+          <Form>
+            <Field>
+              <Label>Name</Label>
+
+              <Control>
+                <Input type="text" placeholder="Text input" />
+              </Control>
+            </Field>
+
+            <Field>
+              <Label>Username</Label>
+
+              <Control>
+                <Input
+                  color="success"
+                  type="text"
+                  placeholder="Text input"
+                  defaultValue="trunx"
+                />
+
+                <Help color="success">This username is available</Help>
+              </Control>
+            </Field>
+
+            <Field>
+              <Label>Email</Label>
+
+              <Control>
+                <Input
+                  color="danger"
+                  type="email"
+                  placeholder="Email input"
+                  defaultValue="hello@"
+                />
+
+                <Help color="danger">This email is invalid</Help>
+              </Control>
+            </Field>
+
+            <Field>
+              <Control>
+                <Checkbox>
+                  <span className={bulma("pl-2")}>
+                    I agree to the <a href="#">terms and conditions</a>
+                  </span>
+                </Checkbox>
+              </Control>
+            </Field>
+
+            <Field>
+              <Control>
+                <Radio name="question">
+                  <span className={bulma("px-1")}>Yes</span>
+                </Radio>
+
+                <Radio name="question">
+                  <span className={bulma("px-1")}>No</span>
+                </Radio>
+              </Control>
+            </Field>
+
+            <Field isGrouped>
+              <Control>
+                <Button color="link">Submit</Button>
+              </Control>
+
+              <Control>
+                <Button color="link" isLight>
+                  Cancel
+                </Button>
+              </Control>
+            </Field>
+          </Form>
+        </Column>
+
+        <Column size="half">
+          <Code
+            snippet={`
+          <Form>
+            <Field>
+              <Label>Name</Label>
+
+              <Control>
+                <Input type="text" placeholder="Text input" />
+              </Control>
+            </Field>
+
+            <Field>
+              <Label>Username</Label>
+
+              <Control>
+                <Input
+                  color="success"
+                  type="text"
+                  placeholder="Text input"
+                  defaultValue="trunx"
+                />
+
+                <Help color="success">This username is available</Help>
+              </Control>
+            </Field>
+
+            <Field>
+              <Label>Email</Label>
+
+              <Control>
+                <Input
+                  color="danger"
+                  type="email"
```

Comment:
Same comment as above. At least for Input we can try and check the result.

Reply:
I remember I tried something similar and it works.
Also there are few helpers that were added recently, you can do some things like

```ts
import { Form, FormOnSubmit, formValues } from "trunx"

const fields = ["name"];
type Field = (typeof fields)[number];
const fieldName = {
  foo: "foo",
} as const satisfies Record<string, Field>;

const MyForm = () => {
  const onSubmit = useCallback<FormOnSubmit>(
    (event) => {
      event.preventDefault();
      if (!canSubmit) return;
      const { foo } = formValues(event, fields);
      // API call here
    },
    [canCreate]
  );

  return (
    <Form onSubmit={onSubmit}>
      <Input name={fieldName.foo} required />
    </Form>
  )
}
```